vllm/model_executor/layers/fused_moe/cutlass_moe.py

```diff
@@ -96,8 +97,14 @@ def cutlass_moe_fp8(
     n = w2_q.size(1)
 
     topk = topk_ids.size(1)
+    assert topk == 1, \
+        "apply_router_weight_on_input is currently only implemented for topk=1"
```

Comment:
Should we move this assert to be in the `if apply_router_weight_on_input:` conditional? This seems restrictive without checking if apply_router_weight_on_input is true